### PR TITLE
SCL-18372 Instruct BSP server to workspace/reload during refresh

### PR DESCRIPTION
**Pull request by retronym**

### CODE REVIEW

File: bsp/src/org/jetbrains/bsp/project/importing/BspProjectResolver.scala [hunk 1]
```diff
@@ -72,7 +72,7 @@ class BspProjectResolver extends ExternalSystemProjectResolver[BspExecutionSetti
     reporter.startTask(structureEventId, None, BspBundle.message("bsp.resolver.resolving.build.structure"))
     val targetsEventId = BuildMessages.randomEventId
     reporter.startTask(targetsEventId, Some(structureEventId), BspBundle.message("bsp.resolver.build.targets"))
-    val targetsRequest = server.workspaceBuildTargets()
+    val targetsRequest = server.workspaceReload().thenCompose(_ => server.workspaceBuildTargets())
```
Comment (retronym) [inline bsp/src/org/jetbrains/bsp/project/importing/BspProjectResolver.scala]: You might want to retain backwards compatibility with BSP servers that don't implement this method. We don't have [protocol version negotiation yet](https://github.com/build-server-protocol/build-server-protocol/issues/56), but I guess we could just swallow a "not-implemented" error here.

Comment (jastice) [reply bsp/src/org/jetbrains/bsp/project/importing/BspProjectResolver.scala]: There's a flag for support of this: `BuildServerCapabilities.canReload`
But I see you're already using it :)

File: project/dependencies.scala [hunk 2]
```diff
@@ -10,7 +10,7 @@ object Versions {
   val bloopVersion = "1.4.8-26-cbddb8ba"
   val zincVersion = "1.4.0-M12"
   val intellijVersion = "212.4486"
-  val bspVersion = "2.0.0-M12+27-4994bd9d-SNAPSHOT"
+  val bspVersion = "2.0.0-M14+7-56ef5d5f-SNAPSHOT"
```
Comment (jastice) [inline project/dependencies.scala]: We can use M14 without the latest snapshot

**Review by jastice — APPROVED** [review]
Thanks! I must have forgotten to add this ^^

File: project/dependencies.scala (hunk 2 above)
Comment (retronym) [reply project/dependencies.scala]:
```suggestion
  val bspVersion = "2.0.0-M14"
```

**Comment by jastice** [comment]
YouTrack issue: https://youtrack.jetbrains.com/issue/SCL-19230

**Pull request merged**